src/many-abci/src/abci_app.rs

```diff
@@ -276,22 +302,36 @@ impl Application for AbciApp {
                     }
                 }
 
+                {
+                    let cache = self.cache.write();
+                    if let Err(_e) = cache {
+                        return ResponseDeliverTx {
+                            code: ManyAbciErrorCodes::FrontendError as u32,
+                            ..Default::default()
+                        };
+                    }
+                    if let Err(e) = cache.unwrap().message_executed(&cose, &response) {
+                        // We log those errors, but the message already executed,
+                        // so we don't return an error.
+                        tracing::error!("message_executed failed: {e}");
+                    }
```

Comment:
So this is an error that happens in ABCI after the backend was successful. For example, a `send` transaction was successful and fully run, but the many-abci bridge itself couldn't save to the database due to I/O. That means that we shouldn't return an error to tendermint (the transaction was fully executed and the app hash was updated), but the state is bad. The only way to move forward properly would be to call the ledger again with a reverse transaction (since that transaction already succeeded, we need to remove it from the ledger state, including events), then return an error to tendermint. But also, that transaction is not a bad one, just something unexpected happened.

This is a nasty problem, and I considered panicking. This should be a sign that you are in an unstable state. I decided that the worst that can happen currently is that the node will lose consensus and we can investigate why, but that in most cases things can continue as they were.

If you want to discuss I'm open to it.

Reply:
Resolved by panicking if `message_executed` fails for some reason.